[yadunut] iP

yadunut opened the pull request:
# Yadunand's IP Project
> Not knowledge, but knowledge of knowledge, heralds the death of ignorance. - Halcyon (CS3234)

Duke is a 
- text based
- 0% AI 🤩

chatbot. 

To install, 
1. Clone this repo [yadunut/ip](github.com/yadunut/ip)
2. Run `git clone https://github.com/yadunut/ip`

There are some unfinished tasks tho
- [ ] Remove **AI**
- [ ] Add GUI


```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

yadunut committed:
feat: Add name of bot

yadunut committed:
feat: add ability to echo back and quit on bye

yadunut committed:
feat: add ability to mark tasks as done and undone

yadunut committed:
feat: add todos, deadlines and tasks

yadunut committed:
feat: add error checking

yadunut committed:
feat: add task deletion

yadunut committed:
feat: add enums to manage todo state

yadunut committed:
chore: add ui tests

yadunut committed:
Add Parser and Writer for writing to files

yadunut committed:
Add the data dir to gitignore

yadunut committed:
Merge branch 'branch-Level-7'

yadunut committed:
Modify Tasks to support dates

yadunut committed:
Move parsing to a utility function

yadunut committed:
Update ui tests with the new features

yadunut committed:
Merge branch 'branch-Level-8'

yadunut committed:
Add Storage Abstraction

yadunut committed:
Add Ui Abstraction

yadunut committed:
Add Abstraction over Tasks

yadunut committed:
Add Commands and the Parser

yadunut committed:
Refactor Duke with the New OOP

yadunut committed:
Merge branch 'A-MoreOOP'

yadunut committed:
Refactor Command to seperate files

yadunut committed:
Refactor Parser to package

yadunut committed:
Refactor Storage

yadunut committed:
Refactor tasks

yadunut committed:
Refactor Ui

yadunut committed:
Refactor Utils

yadunut committed:
Refactor Duke to new packages

yadunut committed:
Move Duke.java into Duke package

yadunut committed:
Merge branch 'A-Packages'

yadunut committed:
Merge remote-tracking branch 'origin/add-gradle-support'

yadunut committed:
Fix adding gradle

yadunut committed:
Add tests for parser

yadunut committed:
Merge branch 'A-JUnit'

yadunut committed:
Add javadoc for the given files

yadunut committed:
Fix naming and final delcaration violations

yadunut committed:
Add command to find tasks

yadunut committed:
Merge branch 'Level-9'

yadunut committed:
Merge branch 'A-JavaDoc'

yadunut committed:
Add checkstyle config files

yadunut committed:
Merge branch 'A-CheckStyle'

4llysa commented on src/main/java/duke/storage/FileStorage.java:
```diff
@@ -0,0 +1,92 @@
+                String[] split = fileScanner.nextLine().split(" \\| ");
+                TodoState state = split[1].equals("1") ? TodoState.DONE : TodoState.UNDONE;
+                switch (split[0]) {
+                    case "T": {
```
Should `case` clauses be indented?

According to the [java conventions](https://se-education.org/guides/conventions/java/intermediate.html) used by this course, you may need to un-indent all of them, alongside the case statements in other files.

4llysa commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,32 @@
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.ui.TextUi;
```
I like how you organised your import statements and imported every class separately. Keep up the good work!

4llysa commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,12 @@
+import duke.task.TaskList;
+import duke.ui.TextUi;
+
+public class ListCommand extends Command {
```
Perhaps you could add some Javadocs for all of your classes? I noticed that many of the Command classes do not have javadocs

4llysa commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,93 @@
+package duke.parser;
+
+import duke.command.*;
```
Not sure I like the use of wildcard imports here. Perhaps you could checkout the settings of your IDE to ensure it does not collapse the import statements into a wildcard automatically for you.

yorklim replied:
I know that importing each class individually can be quite messy but please import each class individually as mentioned by the java coding convention

4llysa commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,93 @@
+                return new ByeCommand();
+            }
+            case "help":
+            // Fallthrough
```
Excellent! I like your use of ` // Fallthrough `, demonstrating your awareness of this potential pitfall

4llysa commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,66 @@
+    public static void main(String[] args) throws IOException {
+        new Duke(DukeConfig.Default()).run();
+    }
+}
```
Overall, I appreciated the very clean and consistent code in your project :)

yorklim commented on src/main/java/duke/command/Command.java:
Love that you broke down the command class further down into subclasses to help with reading and OOP!

yorklim commented on src/main/java/duke/task/TodoState.java:
Good use of enum class to signal status of tasks

yorklim commented on src/main/java/duke/ui/TextUi.java:
```diff
@@ -0,0 +1,80 @@
+import java.util.Scanner;
+
+public class TextUi {
+    private static final String LINE = "____________________________________________________________";
```
Like that you made the line break a constant instead of typing "____' everywhere, makes code cleaner

yorklim commented on src/main/java/duke/task/Task.java:
Can consider being more consistent when using this. , in this class and some other classes, this. is not used in functions when mentioning the variable but in TaskList, this. was used even in functions. Can consider either removing all the this. or add it everywhere

yorklim commented on src/main/java/duke/storage/FileStorage.java:
```diff
@@ -0,0 +1,92 @@
+     * next save.
+     */
+    @Override
+    public List<Task> load() throws IOException, DukeException {
```
Can consider changing the indent of switch. case statements to follow the jave coding convention used by the course

yorklim reviewed (commented):
Overall I like that you code is clean it LGTM, just some minor nits!

yadunut committed:
Add GUI with javaFX

yadunut committed:
Merge branch 'Level-10'

yadunut committed:
Enforce code Quality

Intellij's default formatter is updated to support enforcing SE-EDU Code quality
Standards.

The default formatter's settings was updated with the SE-EDU guide to better format
code.

yadunut committed:
Merge branch 'A-CodeQuality'

yadunut committed:
Add assertions to commands and gui

Add assertions to places where dependencies are injected. This allows us to ensure that

yadunut committed:
Merge branch 'A-Assertions'

yadunut committed:
Add ordering to tasks to support sorting

yadunut committed:
Merge branch 'C-Sort'

yadunut committed:
Refactor duke to Brian

yadunut committed:
More brian refactors

yadunut committed:
Merge branch 'Brian-refactor'

yadunut committed:
Add documentation on usage of AI

yadunut committed:
Proper duke to brian refactor

yadunut committed:
Link textUI to GUI

yadunut committed:
Add User Guide
